File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/components/gesturehandler/OnTouchEventListener.java
package versioned.host.exp.exponent.modules.api.components.gesturehandler;

import android.view.MotionEvent;

import versioned.host.exp.exponent.modules.api.components.gesturehandler.GestureHandler;

public interface OnTouchEventListener<T extends GestureHandler> {
  void onTouchEvent(T handler, MotionEvent event);
  void onStateChange(T handler, int newState, int oldState);
}

File: android/vendored/unversioned/react-native-gesture-handler/android/lib/src/main/java/com/swmansion/gesturehandler/OnTouchEventListener.kt
package com.swmansion.gesturehandler

import android.view.MotionEvent

interface OnTouchEventListener {
  fun <T : GestureHandler<T>> onHandlerUpdate(handler: T, event: MotionEvent)
  fun <T : GestureHandler<T>> onStateChange(handler: T, newState: Int, oldState: Int)
  fun <T : GestureHandler<T>> onTouchEvent(handler: T)
}
